### PR TITLE
Add a docker-stack.yml to support a basic Docker Swarm deployment

### PR DESCRIPTION
**Pull request by hastarin**
NOTE: I only have this running in a test environment and haven't done much with it yet but creating a pull request since @deviantony asked.

Closes #246

### CODE REVIEW

**Comment by deviantony** [comment]
Thanks, I'll try to have a look at this shortly.

File: elasticsearch/config/elasticsearch.yml [hunk 1]
```diff
@@ -13,4 +13,4 @@ discovery.zen.minimum_master_nodes: 1
 ## Use single node discovery in order to disable production mode and avoid bootstrap checks
 ## see https://www.elastic.co/guide/en/elasticsearch/reference/current/bootstrap-checks.html
 #
-discovery.type: single-node
+discovery.type: zen
```
Comment (deviantony) [inline elasticsearch/config/elasticsearch.yml]: @antoineco Do you think that might introduce issues with the standard deployment method (e.g. compose)?

Comment (hastarin) [reply elasticsearch/config/elasticsearch.yml]: Sorry that's a leftover from my local experiments.  I'll remove that.

File: docker-stack.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
+version: '3.3'
+
+networks:
+  net:
```
Comment (deviantony) [inline docker-stack.yml]: For consistency with the compose file, please rename the network to `elk`.

File: docker-stack.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      - source: logstash_pipeline
+        target: /usr/share/logstash/pipeline/logstash.conf
+    environment:
+      LS_JAVA_OPTS: "-Xmx256m -Xms256m"
+    networks:
+      - net
+    deploy:
+      mode: replicated
+      replicas: 1
+      resources:
```
Comment (deviantony) [inline docker-stack.yml]: I don't know if the resources limit are actually required here. This project aims to provide a generic solution, we might want to let the user actually define these.

File: docker-stack.yml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+      - "5601:5601"
+    configs:
+      - source: kibana_config
+        target: /usr/share/kibana/config/kibana.yml
+    networks:
+      - net
+    deploy:
+      mode: replicated
+      replicas: 1
+      resources:
```
Comment (deviantony) [inline docker-stack.yml]: I don't know if the resources limit are actually required here. This project aims to provide a generic solution, we might want to let the user actually define these.

File: docker-stack.yml (hunk 2 above)
Comment (hastarin) [reply docker-stack.yml]: Are you sure?  If you deploy the stack as "elk" you will then end up with a network named "elk_elk".

File: docker-stack.yml (hunk 3 above)
Comment (hastarin) [reply docker-stack.yml]: You have JAVA limitations passed in via environment variables.  Wouldn't it make sense to provide something similar for the docker container itself?   Users can easily edit or remove these but if you're new to docker as I am you may not even know they exist without seeing them there.

**Comment by hastarin** [comment]
I know next to nothing about Travis CI.   NOTE:  Even locally using "localhost" does not work for me.  Using the hostname or 127.0.0.1 does.  If this latest build doesn't pass I don't know what more I can try without spending more time than I have available on it.

File: docker-stack.yml (hunk 2 above)
Comment (deviantony) [reply docker-stack.yml]: Sorry for the late update @hastarin 

Yes, it's all right, please rename it to `elk`.

**Comment by deviantony** [comment]
@antoineco any hint on the Travis issue?

**Comment by antoineco** [comment]
@deviantony looks like a benign timeout issue to me, Swarm may take slighly longer than Compose to start services, let me check.

**Comment by antoineco** [comment]
That was it. I increased the sleep to 40sec and the test plan succeeds 👍

**Pull request closed without merging**